src/ProductTour/Checkpoint.jsx

```diff
@@ -0,0 +1,142 @@
+      const checkpointPopper = createPopper(targetElement, checkpoint, {
+        placement: isMobile ? 'top' : placement,
+        modifiers: [
+          {
+            name: 'arrow',
+            options: {
+              padding: 25,
+            },
+          },
+          {
+            name: 'offset',
+            options: {
+              offset: [0, 20],
+            },
+          },
+          {
+            name: 'preventOverflow',
+            options: {
+              padding: 20,
+              tetherOffset: 35,
+            },
+          },
+        ],
+      });
+      setCheckpointVisible(true);
+      if (checkpointPopper) {
+        checkpointPopper.forceUpdate();
+      }
+    }
+  }, [target, isMobile]);
+
+  useEffect(() => {
+    if (checkpointVisible) {
+      // Scroll the Checkpoint into view once its rendered
+      const targetElement = document.querySelector(target);
+      let targetOffset = targetElement.getBoundingClientRect().top;
+      if ((targetOffset < 0) || (targetElement.getBoundingClientRect().bottom > window.innerHeight)) {
+        if (placement.includes('top')) {
+          if (targetOffset < 0) {
+            targetOffset *= -1;
+          }
+          targetOffset -= 280;
+        } else {
+          targetOffset -= 80;
+        }
+
+        window.scrollTo({
+          top: targetOffset, behavior: 'smooth',
+        });
+      }
+
+      const button = document.querySelector('#pgn__checkpoint-primary-button');
+      button.focus();
+    }
+  }, [target, checkpointVisible]);
+
+  const isLastCheckpoint = index + 1 === totalCheckpoints;
+  const isOnlyCheckpoint = totalCheckpoints === 1;
+
+  return (
+    <div
+      id="checkpoint"
+      className="pgn__checkpoint p-4 shadow-lg"
+      aria-labelledby="pgn__checkpoint-title"
+      role="dialog"
+      style={{ visibility: checkpointVisible ? 'visible' : 'hidden', pointerEvents: checkpointVisible ? 'auto' : 'none' }}
+    >
+      {/* This text is not translated due to Paragon's lack of i18n support */}
+      <span className="sr-only">Top of step {index + 1}</span>
+      {(title || !isOnlyCheckpoint) && (
+        <div className="d-flex justify-content-between mb-2.5">
+          <CheckpointTitle>{title}</CheckpointTitle>
+          <CheckpointBreadcrumbs currentIndex={index} totalCheckpoints={totalCheckpoints} />
+        </div>
+      )}
+      <CheckpointBody>{body}</CheckpointBody>
+      <CheckpointActionRow
+        isLastCheckpoint={isLastCheckpoint}
+        {...props}
+      />
+      <div id="pgn__checkpoint-arrow" data-popper-arrow />
+      {/* This text is not translated due to Paragon's lack of i18n support */}
+      <span className="sr-only">Bottom of step {index + 1}</span>
+    </div>
+  );
+}
+
+Checkpoint.defaultProps = {
+  advanceButtonText: null,
+  body: null,
+  dismissButtonText: null,
+  endButtonText: null,
+  placement: 'top',
+  title: null,
+};
+
+Checkpoint.propTypes = {
+  advanceButtonText: PropTypes.node,
```

Comment:
Okay! I didn't initially add comments here because this component isn't (currently) exposed on the doc site but I'm happy to do so incase we decide to expose it